### ISSUE TITLE
`import/extensions` not flagging missing extension on TypeScript path imports

### ISSUE THREAD

**Issue by nemanjastanic** [import-js/eslint-plugin-import#2955]
I'm trying to enforce an extension to be added when using imports, I only have one problem left here.

This will not be enforced:
```ts
import a from "@a/file";
```

Here is my config:
```cjs
module.exports = {
  extends: ["turbo", "plugin:import/recommended"],
  parser: "@typescript-eslint/parser",
  env: { node: true, es6: true },
  parserOptions: {
    ecmaVersion: "latest",
    sourceType: "module",
  },
  settings: {
    "import/resolver": {
      typescript: { alwaysTryTypes: true, extensions: [".js"] },
      node: { extensions: [".js"] },
    },
  },
  rules: {
    "no-console": ["error", { allow: ["info", "warn", "error"] }],
    "turbo/no-undeclared-env-vars": "off",
    "import/extensions": [
      "error",
      "always",
      { json: "always", js: "always", ts: "never" },
    ],
  },
};
```

What am I missing here or is this a limitation?

**Comment by ljharb** [comment]
Extensions don't apply to things that aren't relative paths - `@a/file` is a package, and it wouldn't work if you appended an extension.

**Comment by nemanjastanic** [comment]
> Extensions don't apply to things that aren't relative paths - `@a/file` is a package, and it wouldn't work if you appended an extension.

Like I mentioned, it is a TypeScript path, it's not a package.

Even so, why is there an option for `ignorePackages` as the second argument of the rule then?

Node does force you to define extensions for packages as well, eg. `dayjs/plugin/utc.js`.

**Comment by ljharb** [comment]
That’s not node, that’s up to the package - and a package that has chosen an extension for an entry point has failed at DX :-)

ignorePackages should be true for everyone, it’s just legacy that keeps the default what it is.

I strongly suggest not using @ for aliases, since that character is already used by npm scopes. What happens if you use a different character in the TS alias?

**Comment by ljharb** [comment]
@nemanjastanic does it work if you use a different character?
